scarb/src/compiler/plugin/proc_macro/compilation.rs

```diff
@@ -26,3 +32,52 @@ impl SharedLibraryProvider for Package {
             .into_child(lib_name)
     }
 }
+
+pub fn compile_unit(unit: CompilationUnit, ws: &Workspace<'_>) -> Result<()> {
+    let main_package = unit.components.first().unwrap().package.clone();
+
+    let mut cmd = CargoCommand::builder()
+        .current_dir(main_package.root().to_path_buf())
+        .target_dir(
+            main_package
+                .target_path(ws.config())
+                .path_unchecked()
+                .to_path_buf(),
+        )
+        .build();
+
+    {
+        let _ = trace_span!("compile_proc_macro").enter();
+        let status = cmd
+            .status()
+            .with_context(|| format!("Failed to execute {:?}", cmd))?;
+        if !status.success() {
+            return Err(anyhow::anyhow!(
+                "Failed to compile procedural macro plugin: {:?}",
+                cmd
+            ));
+        }
```

Comment:
1. errors start with lowercase
2. why not context chaining?
3. consider using `ensure!`